gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractRequest.java

```diff
@@ -197,15 +196,20 @@ public WebSocket webSocket() {
 
     @Override
     public MutableRequest contextPath(String contextPath) {
+        this.contextPath = contextPath;
+        this.pathInfo = path().substring((contextPath.length() == 1) ? 0 : contextPath.length() - 1);
+        return this;
```

Comment:
What about keeping the null check of contextPath before change pathInfo ? 

```suggestion
        this.contextPath = contextPath;
        
        if (contextPath == null || contextPath.isEmpty()) {
            this.pathInfo = path();
        } else if (path().length() > 1) {
            this.pathInfo = path().substring(contextPath.length() - 1);
        } else {
            this.pathInfo = "";
        }
        
        return this;
```

Reply:
Done